src/IECoreVDB/VDBScene.cpp

```diff
@@ -180,6 +180,31 @@ class VDBScene : public SceneInterface
 			throw IECore::NotImplementedException("");
 		}
 
+
+		////////////////////////////////////////////////////
+		// Sets
+		////////////////////////////////////////////////////
+
+		NameList setNames() const override
+		{
+			return NameList();
+		}
+
+		IECore::ConstPathMatcherDataPtr readSet( const Name &name ) const override
+		{
+			return new IECore::PathMatcherData();
+		}
+
+		void writeSet( const Name &name, const IECore::PathMatcherData *set ) override
+		{
+			throw IECore::NotImplementedException( "VDBScene::writeSet() Not supported" );
+		}
+
+		void hashSet( const Name &setName, IECore::MurmurHash &h ) const override
+		{
+			h.append( fileName() );
```

Comment:
Ah, I should have thought of this earlier. All other hashes (Object, ComputeNode, `SceneInterface::hash()`) are first initialised by appending the TypeId of the thing in question, to avoid collisions between hashes generated by distinct types. So, in this case `SceneInterface::hashSet()` should remain pure virtual, but should have an implementation that just does `h.append( typeId() )`. Then all derived classes should first call the base class and then append anything they need.

Once that's done we could omit the fileName here because the inclusion of the TypeId will keep the hash distinct from any other SceneInterface that doesn't implement sets.

Does that make sense?

Reply:
Great. Thanks for the explanation. Just one question... 

> So, in this case SceneInterface::hashSet() should remain pure virtual, but should have an implementation that just does h.append( typeId() )

I'm a little confused if the SceneInterface::hashSet() remains pure virtual then where should the h.append( typeId() ) go?

Reply:
Maybe pure virtual isn't quite the right terminology...this is what I meant...what would you call that?

```
SceneInterface
{
    virtual void hashSet( name, h ) = 0; // Force derived classes to provide an override
}

SceneInterface::hashSet( name, h ) // But still provide a base class implementation they can call
{
     h.append( typeId() );
}

VDBScene::hashSet( name, h )
{
    SceneInterface::hashSet( name, h ); // Call base class from derived class
}
```

Reply:
> what would you call that?

*pure virtual* and I'm an idiot.

```
class Foo 
{
virtual void test() = 0; // I've always assumed this means derived class must implement & there can't be implementation in Foo !!! 
};
```

Thanks for spelling it out.